bot/exts/backend/error_handler.py

```diff
@@ -98,6 +106,42 @@ async def on_command_error(self, ctx: Context, e: errors.CommandError) -> None:
             # ExtensionError
             await self.handle_unexpected_error(ctx, e)
 
+    @Cog.listener("on_app_command_error")
+    async def on_app_command_error(self, interaction: Interaction, e: app_commands.AppCommandError) -> None:
+        """
+        Error handler for app commands.
+
+        See on_command_error for detailed description.
+        """
+        await interaction.response.defer(ephemeral=True)
+
+        debug_message = (
+            f"Command {interaction.command} invoked by {interaction.user} with error "
+            f"{e.__class__.__name__}: {e}"
+        )
+
+        if isinstance(e, app_commands.TransformerError):
+            log.debug(debug_message)
+            self.bot.stats.incr("errors.transformer_error")
+            embed = self._get_error_embed("Transformer error", e.__cause__)
+        elif isinstance(e, app_commands.CommandInvokeError):
+            log.debug(debug_message)
+            self.bot.stats.incr("errors.appcommand_invoke_error")
+            embed = self._get_error_embed("App command invoke error", e.original)
+        else:
+            cause = e.__cause__
```

Comment:
The `cause` here is not really guaranteed to have value.
For example, if I remove the part where it checks it's a TranformerError & throw one explicitly, the cause will be None.

What you can do instead is check whether it's none or not, otherwise use the original.

Reply:
`e.original` is only present for `app_commands.CommandInvokeError` iirc. Also the cause is the original statement that caused the error, eg:

```py
# this happens internally somewhere(not in a transformer or a command)
raise ValueError("a is smaller than b")
```

The cause will be returned as "a is smaller than b".


The only times it retuns `None` is when you explicitly raise an app_command error.

Reply:
Sure, I was referring to the case when we'd raise an app command error explicitly. As I said before, I don't like raising a TextCommand error from within a slash command context

The cause should be present when an exception if raised from another.

```py
try:
    raise ValueError("initial")
except ValueError as e:
    raise KeyError("second") from e
```

when you catch that later, 

```py
try:
    ...
except KeyError as e:
    print(e.__cause__)
    # this will print: initial
````

Reply:
I don't think app command errors are meant to be raised explicitly, I maybe wrong.

Sorry, I don't understand the second part of your comment. Can you elaborate?

Reply:
Let's go with raising a ValueError, which will then become TransformerError, and catch it here

Reply:
Updated in https://github.com/python-discord/bot/pull/2446/commits/6059dd077468b1338109327793400469ed96096e